packages/@webex/plugin-cc/src/AgentProfileService/AgentProfileService.ts

```diff
@@ -0,0 +1,146 @@
+   * @param {string} desktopProfileId ID of the Desktop Profile to be retrieved.
+   * @returns {Promise<DesktopProfileResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async retrieveDesktopProfileById(
+    orgId: string,
+    desktopProfileId: string
+  ): Promise<DesktopProfileResponse> {
+    try {
+      if (!orgId || !desktopProfileId)
+        Promise.reject(new Error('orgId or desktopProfileId is undefined'));
+
+      const URL = `${this.wccAPIURL}organization/${orgId}/agent-profile/${desktopProfileId}`;
+
+      return Promise.resolve(await this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling retrieveDesktopProfileById API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get List of Teams.
+   * @param {string} orgId The ID of an Organization.
+   * @param {number} page (Optional) Index of the page of results to be fetched.
+   * @param {number} pageSize (Optional) Number of items to be displayed on a page.
+   * @param {Array<String>} filter (Optional) Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes (Optional) Specify the attributes to be returned. Supported attributes are id, name, active and workTypeCode.
+   * @returns {Promise<ListTeamsResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfTeams(
+    orgId: string,
+    page?: number,
+    pageSize?: number,
+    filter?: string[],
+    attributes?: string[]
+  ): Promise<ListTeamsResponse> {
+    try {
+      if (!orgId) Promise.reject(new Error('orgId is undefined'));
+      const URL = `${this.wccAPIURL}organization/${orgId}/team?page=${page}&pageSize=${pageSize}&filter=id=in=${filter}&attributes=${attributes}`;
+
+      return Promise.resolve(this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling getListOfTeams API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get List of AuxCodes.
+   * @param {string} orgId The ID of an Organization.
+   * @param {number} page (Optional) Index of the page of results to be fetched.
+   * @param {number} pageSize (Optional) Number of items to be displayed on a page.
+   * @param {Array<String>} filter (Optional) Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes (Optional) Specify the attributes to be returned. Supported attributes are id, name, active and workTypeCode.
+   * @returns {Promise<ListAuxCodesResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfAuxCodes(
+    orgId: string,
+    page?: number,
+    pageSize?: number,
+    filter?: string[],
+    attributes?: string[]
+  ): Promise<ListAuxCodesResponse> {
+    try {
+      if (!orgId) Promise.reject(new Error('orgId is undefined'));
```

Comment:
same comment here as in the other functions

Reply:
Addressed!